src/ToolInfo.php

```diff
@@ -26,67 +26,92 @@
     const COMPOSER_INSTALLED_FILE = '/../../composer/installed.json';
     const COMPOSER_PACKAGE_NAME = 'friendsofphp/php-cs-fixer';
 
-    public static function getComposerVersion()
+    /**
+     * @var string
+     */
+    private $script;
+
+    /**
+     * @var null|string
+     */
+    private $composerVersion;
+
+    /**
+     * @var null|bool
+     */
+    private $isInstalledAsPhar;
+
+    /**
+     * @var null|bool
+     */
+    private $isInstalledByComposer;
+
+    /**
+     * @var null|string
+     */
+    private $scriptDir;
+
+    /**
+     * @param null|string $script
+     */
+    public function __construct($script)
     {
-        static $result;
+        $this->script = (null !== $script ? $script : $_SERVER['SCRIPT_NAME']);
```

Comment:
IMHO this:
```php
$this->script = null !== $script ? $script : $_SERVER['SCRIPT_NAME'];
```
Is unreadable, [operator precedence](http://php.net/manual/en/language.operators.precedence.php) can be a nightmare without parenthesis.